conversant/prompt_chatbot.py

```diff
@@ -186,20 +187,42 @@ def _update_max_context_examples(
             "tokens"
         )
 
-    def reply(self, query: str) -> Interaction:
-        """Replies to a query given a chat history.
+    def should_stop(self, response: str) -> str:
+        """Given a response, returns the stop sequence it ends with if any.
 
-        The reply is then generated directly from a call to a LLM.
+        Args:
+        response (str): Response coming from prompt chatbot.
+
+        Returns:
+        stop_seq (str): The stop sequence at the end of response.
+
+        """
+        for stop_seq in self.client_config["stop_sequences"]:
+            if response.endswith(stop_seq):
+                return stop_seq
+
+        # If there is not a stop sequence at the end
+        return None
+
+    def get_max_reruns(self) -> int:
```

Comment:
I think it would be wise to have a more descriptive name for this function, like ` partial_reply_max_reruns`. Reason: if it will be invoked outside of the `PromptChatbot` class, `get_max_reruns` doesn't specify what it does on first glance